clients/src/main/java/org/apache/kafka/clients/consumer/internals/FetchMetricsManager.java

```diff
@@ -197,22 +274,34 @@ private static String partitionRecordsLagMetricName(TopicPartition tp) {
         return tp + ".records-lag";
     }
 
+    private static String deprecatedMetricName(String name) {
+        return name + ".deprecated";
+    }
+
+    private static boolean shouldReportDeprecatedMetric(String topic) {
+        return topic.contains(".");
+    }
+
     private MetricName partitionPreferredReadReplicaMetricName(TopicPartition tp) {
+        Map<String, String> metricTags = mkMap(mkEntry("topic", tp.topic()), mkEntry("partition", String.valueOf(tp.partition())));
+        return this.metrics.metricInstance(metricsRegistry.partitionPreferredReadReplica, metricTags);
+    }
+
+    @Deprecated
```

Comment:
We should never deprecate something without including a @deprecated javadoc with the details I outlined in a separate comment.

Reply:
Sure, I have added details here https://github.com/apache/kafka/pull/18232#discussion_r1900780156, please let me know your thoughts.